#48 Filter Component

Pink-Mallow opened the pull request:
#57 Filters Component

## What does this change?

Resolves issue 
- [736ace3](https://github.com/fdnd-agency/designingforrecognition/commit/736ace36313e0a5f6f821fd92757256823b54ee7)
- [bfbf21c](https://github.com/fdnd-agency/designingforrecognition/commit/bfbf21cb548fbad3c79ca4c80c244b3e561c950c)
- [85810d7](https://github.com/fdnd-agency/designingforrecognition/commit/85810d7e3e93e45c501d43ff80632500b07ed15a)
- [4391ded](https://github.com/fdnd-agency/designingforrecognition/commit/4391ded8a12751526a1f0c961039ef62d964a813)

<!-- A PR should have enough detail to be understandable far in the future. e.g what is the problem/why is the change needed, how does it solve it and any questions or points of discussion. Prefer copying information from a GitHub issue over linking to it; the card may not always exist and reviewers may not have access to the board. -->

The filter component has been styled according to the Figma design. I worked HTML first and then worked with CSS styling.

## How Has This Been Tested?
<!-- Link to test results in the Wiki-->

- [ ] [User test]()
- [ ] [Accessibility test]()
- [ ] [Performance test]()
- [ ] [Responsive Design test]()
- [ ] [Device test]()
- [ ] [Browser test]()

## Images

<!-- Usually only applicable to UI changes, what did it look like before and what will it look like after? -->
### this is how it looks with HTML only
<img width="452" height="554" alt="Image" src="https://github.com/user-attachments/assets/58f2b725-dda2-4207-aed8-5237a0aeca27" />

### this is how it looks with CSS (this is mobile version)

<img width="322" height="718" alt="image" src="https://github.com/user-attachments/assets/609bf719-32ac-420d-8caf-411d649e8e27" />

### This is how it looks when it is active (when  a user clicks on it)
<img width="360" height="393" alt="Image" src="https://github.com/user-attachments/assets/00b7065c-7090-4599-97dd-11bb9c3d919a" />

## How to review

<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->
- Can you check if the code is correct and follows the FDND conventions?
- Can you check if it is responsive?
- Can you check if the styling follows the Figma design?
- Can you check if it is accessible through tab?

Pink-Mallow committed:
#48 #57   HTML First

Writing Filter Semantics first to make sure it is correct. A bit of CSS to give them a subtle layout

Pink-Mallow committed:
#48 #57 HTML First

fix: label/id mixups and improve form setup

I finally figured out how labels and inputs actually connect 😅
- Fixed mismatched label "for" and input "id" pairs
- Gave everything consistent ids (like participation-select)
- Added required + empty default options
- Lowercased values for consistency
- Added some custom select styling (with a little arrow!)
- Improved focus states and spacing for accessibility
- Cleaned up class names and grouping
- Got keyboard navigation working better

Note to future me:
The label's "for" has to exactly match the input's id, and inputs need unique ids.
Also, the CSS uses `input:checked + label`, so the input has to come right before the label (or be wrapped inside it).

Pink-Mallow committed:
#46 #57 HTML

I wrote the filters shorter in HTML

Pink-Mallow committed:
#48 #57 CSS Styling

CSS styling:
i have given it a hover and active state with the styleguide colors according to Figma design.

Pink-Mallow committed:
#48 I added responsiveness

I made sure it is responsive with display flex by making sure i can resize it with different screen width with @mediaquery.

Pink-Mallow committed:
#48 #57 Added styling with input checked

i added a checked in css so when the user chooses a filter, it has a bit of 3d active stylish look.

Pink-Mallow committed:
#57

Pink-Mallow committed:
#57 small code adjustment

Pink-Mallow committed:
#48 #57 Added a :focus so it is keyboard accessible

vsheo commented on src/lib/components/Filters.svelte:
```diff
@@ -0,0 +1,113 @@
+    <input type="checkbox" id="filter-methodes" name="projects" value="Methodes" alt="filter op Methodes" />
+    <label for="filter-methodes">Methodes</label>
+  </div> 
+</section>
```
This section should be changed to a [form](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/form), because the selected filters need to be sent to the server.
That’s only possible when using a form element.

Pink-Mallow replied:
I have changed it (see commit [4b52172](https://github.com/fdnd-agency/designingforrecognition/pull/90/commits/4b52172f765d9ed57cf9d0a8add8282d3fd31531) (latest commit removed the commented part [a3995df](https://github.com/fdnd-agency/designingforrecognition/pull/90/commits/a3995df48420c1ac34c8fcaccdc41a4d9f1bf842)))

vsheo reviewed (changes requested):
the active state looks great, but is the box shadow stays after I remove the filter, until i klik something else

[2025-11-12 10-52-42.webm](https://github.com/user-attachments/assets/831cbfdd-f3c1-4153-b792-f1d4d44028e8)

vsheo commented:
Can you check if the code is correct and follows the FDND conventions?
You are following the FDND conventions

Can you check if it is responsive?
the filters are responsive

Can you check if the styling follows the Figma design?
there is a slight difference with the bordfer radius, but I think this look good aswell

Can you check if it is accessible through tab?
it is possible to navigate through the filters with tab, and activate a filter with space-bar

Pink-Mallow commented:
- [x]  check border radius according to Figma design
- [x] `<form>` needs to be around the filters instead of `<section>` as the selected filters needs to be sent to the serve
- [x]  the filter focus, hover and active needs to be adjusted to make the selecting of the filters user friendlier

Pink-Mallow commented:
# Border radius

@vsheo I have made the Border Radius according to the styling of the list of items of your ProjectCard component. But if you want me to make it strictly according to the Figma design, then i will.

vsheo commented:
> # Border radius
> @vsheo I have made the Border Radius according to the styling of the list of items of your ProjectCard component. But if you want me to make it strictly according to the Figma design, then i will.

no i think this looks better

Pink-Mallow committed:
#48 #58 Change of code

I changed `<section>` into `<form>` so the filters will be sent to the server. But since we can't sent it to the Directus yet, it will remain this way.

Pink-Mallow commented:
# Change `<section>` to `<form>`

I have changed the section to form (see commit [4b52172](https://github.com/fdnd-agency/designingforrecognition/pull/90/commits/4b52172f765d9ed57cf9d0a8add8282d3fd31531))

Mikiyas-hs commented on src/lib/components/Filters.svelte:
```diff
@@ -0,0 +1,113 @@
+    background: var( --color-accent-primary);
+    border-color: var( --color-accent-primary);
+    transition: ease-in 0.3s;
+  }
```
Ik zag op de video dat de active state aan blijft totdat je op een andere button klikt dit komt door dit stukje in je code je hebt daar checked staan en volgens mij zorgt die checked ervoor dat pas als je op de volgende button klikt de active state bij de andere weg gaat omdat je dan een andere button "checked"

Pink-Mallow replied:
I have fixed it (see commit [0c71fd4](https://github.com/fdnd-agency/designingforrecognition/pull/90/commits/0c71fd46434d592d2ace9c16787aa3e3eddf5265)). The reason why it stayed checked was because i should've removed the `:checked` in that piece of code. Now it should work.

Pink-Mallow committed:
#48 #57 Fixed the input checked

I had :checked double in the code. Hence why it didn't worked. Now if you select a filter or unselect it, you can clearly know that it is unselected.

Pink-Mallow commented:
# Code change
Only `:checked` needed to be removed in the `:hover` piece of CSS code.

Pink-Mallow committed:
#48 small refactor

vsheo committed:
Merge branch 'dev' into 48-filter

vsheo committed:
filter component in dev layout
